controller/volume_controller.go

```diff
@@ -152,14 +152,17 @@ func NewVolumeController(
 	}
 	c.cacheSyncs = append(c.cacheSyncs, ds.ReplicaInformer.HasSynced)
 
-	if _, err = ds.ShareManagerInformer.AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
-		AddFunc:    c.enqueueVolumesForShareManager,
-		UpdateFunc: func(old, cur interface{}) { c.enqueueVolumesForShareManager(cur) },
-		DeleteFunc: c.enqueueVolumesForShareManager,
+	if _, err = ds.PodInformer.AddEventHandlerWithResyncPeriod(cache.FilteringResourceEventHandler{
+		FilterFunc: isShareManagerPod,
+		Handler: cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.enqueueVolumesForShareManager,
+			UpdateFunc: func(old, cur interface{}) { c.enqueueVolumesForShareManager(cur) },
+			DeleteFunc: c.enqueueVolumesForShareManager,
+		},
 	}, 0); err != nil {
 		return nil, err
 	}
-	c.cacheSyncs = append(c.cacheSyncs, ds.ShareManagerInformer.HasSynced)
+	c.cacheSyncs = append(c.cacheSyncs, ds.PodInformer.HasSynced)
```

Comment:
Can you elabore more on the purpose of the change?

Reply:
Looks like it is to reduce the unnecessary enqueue by adding the filter `isShareManagerPod`

Also use `ds.PodInformer.HasSynced` instead of the old `ds.ShareManagerInformer.HasSynced` because we are using podInfomer. Looks like the previous `ds.ShareManagerInformer.HasSynced` was a bug

Reply:
We fixed it at https://github.com/longhorn/longhorn-manager/pull/2811#issuecomment-2243781157. Thank you